File: src/Token.java
import java.util.ArrayList;

public class Token {


    public static final int ERROR = 0;
    public static final int IDENTIFIER = 1; // id
    public static final int ARITMETHIC = 2; // +, -, *, /, ^
    public static final int RELATIONAL = 3; // >, <, ==
    public static final int LOGICAL = 4; // &, |, !
    public static final int ASSIGMENT = 5; // =
    public static final int PUNCTUATION = 6; // =
    public static final int LOGICO = 7;
    public static final int KEYWORD = 8;
    public static final int ENTERO = 9;
    public static final int REAL = 10;

    private int tokenType;
    private String tokenValue;
    private int line;
    private int character;

    public Token(int tokenType, String tokenValue, int line, int character) {
        this.tokenType = tokenType;
        this.tokenValue = tokenValue;
        this.line = line;
        this.character = character;
        //System.out.println(this.toString());

    }

    public String toString() {
        return "Type: " + this.tokeTypeStr() + " Val: " + this.tokenValue + " Line: " + this.line + " Character Pos: " + this.character;
    }

    private String tokeTypeStr() {
        String type;
        switch (this.tokenType) {
            case ERROR:
                type = "ERROR";
                break;
            case IDENTIFIER:
                type = "IDENTIFIER";
                break;
            case ARITMETHIC:
                type = "ARITMETHIC";
                break;
            case RELATIONAL:
                type = "RELACIONAL";
                break;
            case LOGICAL:
                type = "LOGICAL";
                break;
            case PUNCTUATION:
                type = "PUNCTUATION";
                break;
            case KEYWORD:
                type = "KEYWORD";
                break;
            case LOGICO:
                type = "LOGICO";
                break;
            case REAL:
                type = "REAL";
                break;
            case ENTERO:
                type = "ENTERO";
                break;
            default:
                type = "ERROR";
                break;
        }
        return type;
    }

}

File: token.py
ERROR = 0
KEYWORD = 1
IDENTIFIER = 2
LOGICAL = 3
RELATIONAL = 4
ARITMETHIC = 5
PUNCTUATION = 6
INTEGER = 7
REAL = 8
ASSIGMENT = 9


def build(type, val, line, charPos) :
  return {"type":type, "val":val, "line":line, "char":charPos}

def string(token) :
  tokenDic = {
    0:'ERROR',
    1:'KEYWORD',
    2:'IDENTIFIER',
    3:'LOGICAL',
    4:'RELATIONAL',
    5:'ARITMETHIC',
    6:'PUNCTUATION',
    7:'INTEGER',
    8:'REAL',
    9:'ASSIGMENT'
  }
  return ("Type " + str(tokenDic.get(token.get("type"))) + " Val '" + str(token.get("val")) + 
          "' Line " + str(token.get("line")) + " Char Pos " + str(token.get("char")))